adapters/logan/logan.go

```diff
@@ -0,0 +1,147 @@
+package logan
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/buger/jsonparser"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	LoganBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var err error
+	var adapterRequests []*adapters.RequestData
+
+	var originalImpSlice []openrtb2.Imp = request.Imp
+
+	for i := range originalImpSlice {
+		request.Imp = []openrtb2.Imp{originalImpSlice[i]}
+
+		temp := reqBodyExt{LoganBidderExt: reqBodyExtBidder{}}
+		temp.LoganBidderExt.PlacementID, err = jsonparser.GetString(request.Imp[0].Ext, "bidder", "placementId")
+		if err != nil {
+			return nil, []error{err}
+		}
+		temp.LoganBidderExt.Type = "publisher"
+
+		finalyImpExt, err := json.Marshal(temp)
+		if err != nil {
+			return nil, []error{err}
+		}
```

Comment:
Does this look better to you(instead of L43-L55):
```
var errors []error
....


                currImp := originalImpSlice[i]
		request.Imp = []openrtb2.Imp{currImp}

		var bidderExt reqBodyExt
		if err := json.Unmarshal(currImp.Ext, &bidderExt); err != nil {
			errors = append(errors, &errortypes.BadInput{
				Message: err.Error(),
			})
			continue // or return
		}

		bidderExt.LoganBidderExt.Type = "publisher" // constant

		finalImpExt, err := json.Marshal(bidderExt)
		if err != nil {
			return nil, []error{err}
		}
``` 
This is a common approach for other bidders. Unit tests pass with this change. 

Please extract "publisher" to a constant.